hebiros_gazebo_plugin/plugin/hebiros_gazebo_plugin.cpp

```diff
@@ -1,7 +1,54 @@
 #include <hebiros_gazebo_plugin.h>
 
+// Checks the content of a string before the first "." at compile time.
+// Necessary because Gazebo only defines string version numbers.
+// Note: This requires -std=c++14 or higher to compile
+constexpr int GetGazeboVersion ( char const* string_ver )
+{
+  int res = 0;
+  int i = 0;
+  while (string_ver[i] != '\0' && string_ver[i] >= '0' && string_ver[i] <= '9') {
+    res *= 10;
+    res += static_cast<int>(string_ver[i] - '0');
+    ++i;
+  }
+  return res;
+}
+
+// This is a templated struct that allows for wrapping some of the Gazebo
+// code for which compilation differse between versions; we use partial
+// template specialization to compile the appropriate version.
+template<int GazeboVersion, class JointType> struct GazeboHelper;
```

Comment:
Two small things here:

1: I never use `class` over `typename` for template parameters. IIRC, you _technically_ need to when doing a "template template" - but that doesn't apply here
2: If you manage to get a system to compile with a gazebo version not 7 or 9, an unreadable compiler error will be raised. We should have a "default" template structure defined for any other version. I know this caused an issue last time we tried it on `gcc` (but not clang I think?), but that's because we didn't define it thoroughly correct. You can do something like this instead:

```c++
template<int V, typename Joint> struct GazeboHelper {
  static_assert(V == 7 || V == 9, "Unknown version of gazebo");
// rest of definition
};
```

Reply:
Good call -- will add this.